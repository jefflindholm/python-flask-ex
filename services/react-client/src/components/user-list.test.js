import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import UserList from './user-list';

const users = [
  {
    active: true,
    email: 'email1@foo.com',
    id: 1,
    username: 'username_1',
  },
  {
    active: true,
    email: 'joes_email@foo.com',
    id: 2,
    username: 'joes_username',
  },
];

test('UserList renders properly', () => {
  const wrapper = shallow(<UserList users={users}/>);
  const element = wrapper.find('h4');
  expect(element.length).toBe(2);
  const expected = [`${users[0].username}`,'(',`${users[0].email}`, ')'];
  const results = element.get(0).props.children;
  expect(JSON.stringify(results)).toBe(JSON.stringify(expected));
});

test('UserList renders snapshot properly', () => {
  const tree = renderer.create(<UserList users={users}/>);
  expect(tree).toMatchSnapshot();
});
