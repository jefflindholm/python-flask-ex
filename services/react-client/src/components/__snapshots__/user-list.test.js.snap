// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserList renders snapshot properly 1`] = `
<div>
  <h4
    className="box title is-4"
  >
    username_1
    (
    email1@foo.com
    )
  </h4>
  <h4
    className="box title is-4"
  >
    joes_username
    (
    joes_email@foo.com
    )
  </h4>
</div>
`;
